src/base/projects/Browser.svelte

```diff
@@ -79,9 +80,15 @@
     // Close mobile tree if user navigates to other file
     mobileFileTree = false;
 
-    if (path) {
-      project.navigateTo({ path: newPath, revision, line: null });
-    }
+    navigate({
+      type: "projects",
+      params: {
+        urn: project.urn,
+        path: newPath,
+        route: null,
+        revision,
+      },
+    });
```

Comment:
This function should just take the route name and params: `navigate("projects", { urn: project.urn, ... })`

Reply:
The interesting part here is to be able to pass a `Route` type directly into `navigate` for type safety at compile time.
If we separate this into two params we would loose it.